### PR TITLE
Fix linking issues for ODE on macOS

### PR DESCRIPTION
**Pull request by nkalupahana**
### Description

This fixes the Bullet linking issues on macOS introduced by #489 (https://github.com/nkalupahana/ros2-foxy-macos/runs/3014655671?check_suite_focus=true). It also fixes an ODE linking issue that I've been using a temp fix for -- OMPL uses ODE, and this needs to be linked in (https://github.com/nkalupahana/ros2-foxy-macos/runs/3022396910?check_suite_focus=true).

(Bullet issue fixed by #530 so now this is just ODE)

Proof of build: https://github.com/nkalupahana/ros2-foxy-macos/runs/3198722057?check_suite_focus=true

### Checklist
- [x] **Required by CI**: Code is auto formatted using [clang-format](http://moveit.ros.org/documentation/contributing/code)
- [ ] Extend the tutorials / documentation [reference](http://moveit.ros.org/documentation/contributing/)
- [ ] Document API changes relevant to the user in the [MIGRATION.md](https://github.com/ros-planning/moveit/blob/master/MIGRATION.md) notes
- [ ] Create tests, which fail without this PR [reference](https://ros-planning.github.io/moveit_tutorials/doc/tests/tests_tutorial.html)
- [ ] Include a screenshot if changing a GUI
- [ ] While waiting for someone to review your request, please help review [another open pull request](https://github.com/ros-planning/moveit/pulls) to support the maintainers

### CODE REVIEW

File: moveit_core/collision_detection_bullet/CMakeLists.txt [hunk 1]
```diff
@@ -13,6 +13,10 @@ set_target_properties(${MOVEIT_LIB_NAME} PROPERTIES VERSION "${${PROJECT_NAME}_V
 ament_target_dependencies(${MOVEIT_LIB_NAME} SYSTEM
   BULLET
 )
+if(APPLE)
+  target_link_directories(${MOVEIT_LIB_NAME} PUBLIC ${BULLET_LIBRARY_DIRS})
```
Comment (henningkayser) [inline moveit_core/collision_detection_bullet/CMakeLists.txt:17]: `ament_target_dependencies` should already search for the correct library dirs. (see [here](https://github.com/ament/ament_cmake/blob/master/ament_cmake_target_dependencies/cmake/ament_target_dependencies.cmake#L116)). Or is it actually missing the `PUBLIC` keyword. I didn't check thoroughly, but Bullet doesn't seem to be part of the public API.

File: moveit_planners/ompl/ompl_interface/CMakeLists.txt [hunk 2]
```diff
@@ -26,6 +26,11 @@ set_target_properties(${MOVEIT_LIB_NAME} PROPERTIES VERSION "${${PROJECT_NAME}_V
 
 find_package(OpenMP REQUIRED)
 
+# Used to link in ODE, an OMPL dependency, on macOS
+if(APPLE)
+  target_link_directories(${MOVEIT_LIB_NAME} PUBLIC ${OMPL_LIBRARY_DIRS})
```
Comment (henningkayser) [inline moveit_planners/ompl/ompl_interface/CMakeLists.txt:31]: Same here. Is ODE not exported correctly?

Comment (nkalupahana) [reply moveit_planners/ompl/ompl_interface/CMakeLists.txt:31]: Looking back at this, I think the issue might be that in moveit_planners/ompl/CMakeLists.txt, ompl is lowercase, but the variable is OMPL_LIBRARY_DIRS -- the code you linked to specifically say that case matters. I'll try it out.

File: moveit_core/collision_detection_bullet/CMakeLists.txt (hunk 1 above)
Comment (nkalupahana) [reply moveit_core/collision_detection_bullet/CMakeLists.txt]: I believe this is an issue with the custom FindBULLET. Removing it fixed this issue. Let me know if that's okay -- it is quite old, and I think if the CI passes, it should be.

File: moveit_planners/ompl/ompl_interface/CMakeLists.txt (hunk 2 above)
Comment (nkalupahana) [reply moveit_planners/ompl/ompl_interface/CMakeLists.txt:31]: Looks like this doesn't work -- I have a feeling ODE just isn't exported correctly, so this is required. Let me know if there's something specific you want to try.

**Comment by nkalupahana** [comment]
@henningkayser Let me know if you want any more changes, or if this is good to go!

**Comment by nkalupahana** [comment]
@henningkayser sorry to keep bothering you on this :) I'd like to get this merged in soon if there are no further issues!

**Comment by nkalupahana** [comment]
@tylerjw I'd love to get this merged in if possible before `foxy` gets branched off as mentioned in #530 -- it's a pretty simple one-file change, and it's essential to get MoveIt2 compiling on macOS.

**Comment by tylerjw** [comment]
@nkalupahana do you know why it isn't passing for foxy in CI?

**Comment by nkalupahana** [comment]
@tylerjw It looks like CI is trying to use the ros2 branch instead of the foxy branch of geometric_shapes, which very recently got split off (https://github.com/ros-planning/geometric_shapes/pull/200). I don't believe this has anything to do with this PR.

**Comment by tylerjw** [comment]
You are correct.  I'm working on a fix for that.  Does this PR fix your issue with building on OSX?  I couldn't tell based on the conversation thread above if this worked for you.

**Comment by nkalupahana** [comment]
@tylerjw yes, it does! https://github.com/nkalupahana/ros2-foxy-macos/runs/3200763303?check_suite_focus=true

**Comment by tylerjw** [comment]
Here is what I'm doing about foxy: https://github.com/ros-planning/moveit2/pull/565

I hope to merge that soon and then we can update this based on that and I'm happy to merge this (as you say it fixes your issue and it doesn't seem to harm anything else).

**Review by henningkayser — APPROVED** [review]

**Comment by henningkayser** [comment]
@nkalupahana sorry for letting this get stale. The change looks fine to me, I filed a related issue at ompl. lgtm after CI succeeds

**Pull request merged**